### PR TITLE
Support envs yaml file by using --env-file option

### PR DESCRIPTION
**Pull request by RazRegev**

### CODE REVIEW

File: tests/test_cli.py [hunk 1]
```diff
@@ -1116,6 +1127,47 @@ def test_run_with_defaults_and_env_from_config_file(self, skipper_runner_run_moc
...
+    @mock.patch('__builtin__.open', mock.MagicMock(create=True))
+    @mock.patch('os.path.exists',
+                mock.MagicMock(autospec=True, return_value=True))
+    @mock.patch('yaml.safe_load', mock.MagicMock(
+        autospec=True,
+        return_value=SKIPPER_CONF_WITH_ENV_FILE
+    ))
+    @mock.patch('skipper.utils.load_yaml_file', mock.MagicMock(
+        autospec=True,
+        return_value=CONFIG_ENV_EVALUATION
```
Comment (eranco74) [inline tests/test_cli.py:1139]: I think you should test:
1. validate new envs from the file.
2. validate envs from the file with overrides in the command line (command line should take precedence)
3. Empty env file
4. malformed env file

File: skipper/cli.py [hunk 2]
```diff
@@ -36,12 +37,17 @@ def cli(ctx, registry, build_container_image, build_container_tag, build_contain
...
-    ctx.obj['env'] = ctx.default_map.get('env', {})
     ctx.obj['containers'] = ctx.default_map.get('containers')
     ctx.obj['volumes'] = ctx.default_map.get('volumes')
     ctx.obj['workdir'] = ctx.default_map.get('workdir')
     ctx.obj['workspace'] = ctx.default_map.get('workspace', None)
     ctx.obj['container_context'] = ctx.default_map.get('container_context')
+
+    env = ctx.default_map.get('env', {})
+    if env_file:
+        env.update(utils.load_yaml_file(env_file))
```
Comment (eranco74) [inline skipper/cli.py:48]: Seems that the env file will override the env in the cmd, it should be the other way around.

**Review by eranco74 — COMMENTED** [review]
Just add some test cases.
Change what takes precedence (cmd trumps file)
and update the readme

File: skipper/cli.py (hunk 2 above)
Comment (eranco74) [reply skipper/cli.py:48]: I think a better implementation will be to pass the env file forward (with --env-file) to doker/podman and let it handle it.

Comment (RazRegev) [reply skipper/cli.py:48]: > I think a better implementation will be to pass the env file forward (with --env-file) to doker/podman and let it handle it.

done, opened https://github.com/Stratoscale/skipper/pull/139

**Comment by RazRegev** [comment]
opened https://github.com/Stratoscale/skipper/pull/139 instead

**Pull request closed without merging**